libs/browser/src/lib/services/browser-micro-sentry-client.ts

```diff
@@ -126,13 +132,22 @@ export class BrowserMicroSentryClient extends MicroSentryClient {
     }
 
     this.extendState({
-      breadcrumbs: [
+      [this.breadcrumbsKeyName]: [
         {
           timestamp: Date.now() / 1_000,
           ...result,
         },
       ],
     });
+
+    const breadcrumbs = this.getKeyState(this.breadcrumbsKeyName);
+
+    if (!!breadcrumbs && (breadcrumbs.length ?? 0) > this.maxBreadcrumbs) {
+      this.setKeyState(
+        this.breadcrumbsKeyName,
+        this.maxBreadcrumbs > 0 ? breadcrumbs.slice(-this.maxBreadcrumbs) : []
+      );
+    }
```

Comment:
Maybe trim breadcrumbs on addition before putting to state?
(if so, we won't need getKeyState method)

also it's possible that we should trim breadcrums in setBreadcrumbs method too

Reply:
added to setBreadcrumbs and refactored a bit